### ISSUE TITLE
xtb-v6.6.1 can not be compiled using CMake + ninja

### ISSUE THREAD

**Issue by foxtran** [grimme-lab/xtb#903]
**Describe the bug**
xtb-v6.6.1 can not be compiled using CMake + ninja

**To Reproduce**
Steps to reproduce the behaviour:

```bash
wget https://github.com/grimme-lab/xtb/archive/refs/tags/v6.6.1.tar.gz
tar -xzvf v6.6.1.tar.gz
cd xtb-6.6.1
cmake -G"Ninja" -Bbuild -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_Fortran_COMPILER=gfortran
```
produces:
```
CMake Error at CMakeLists.txt:127 (add_library):
  The install of the lib-xtb-shared target requires changing an RPATH from
  the build tree, but this is not supported with the Ninja generator unless
  on an ELF-based or XCOFF-based platform.  The
  CMAKE_BUILD_WITH_INSTALL_RPATH variable may be set to avoid this relinking
  step.
```

**Expected behaviour**
Compile xtb using CMake & ninja-build

**Additional context**
I'm using
```
cmake --version
cmake version 3.28.0-rc4

ninja --version
1.11.1.git.kitware.jobserver-1

cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.6 (Ootpa)"
```

**Comment by Albkat** [comment]
Hello, 
the new versions of CMake has the bug object targets, but this problem I haven't encountered yet. 
I would wait until the #900 is merged into main, then try again.

**Comment by foxtran** [comment]
@Albkat, thank you!
Looks like in master branch they fixed mentioned issue in #900 a couple days ago: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/8974
So, cmake 3.28.0-rc5 works properly.

**Comment by foxtran** [comment]
In single-thread mode, one can compile xtb + ninja. Unfortunately, parallel compilation is not yet available. See more here: https://gitlab.kitware.com/cmake/cmake/-/issues/25425
